src/array.cpp

```diff
@@ -77,6 +77,29 @@ namespace sparrow
         return (*this)[size() - 1];
     }
 
+    array array::slice(size_type start, size_type end) const
+    {
+        SPARROW_ASSERT_TRUE(start <= end);
+        array copy = *this;
+        arrow_proxy& arrow_proxy_copy = copy.get_arrow_proxy();
+        arrow_proxy_copy.set_offset(start);
+        arrow_proxy_copy.set_length(end - start);
+        return copy;
+    }
+
+    array array::slice_view(size_type start, size_type end) const
+    {
+        SPARROW_ASSERT_TRUE(start <= end);
+        const arrow_proxy& arrow_proxy_copy = get_arrow_proxy();
+        ArrowSchema* as = new ArrowSchema;
+        *as = arrow_proxy_copy.schema();
+        ArrowArray* ar = new ArrowArray;
+        *ar = arrow_proxy_copy.array();
+        ar->offset = static_cast<int64_t>(start);
+        ar->length = static_cast<int64_t>(end - start);
```

Comment:
This method creates a memory leak, since `as` and `ar` will not be deleted. A way to fix this would be to allocate these structures on the stack and move them in the returned array. The release method should be set to empty methods to avoid destruciton of the data when the view is destroyed.

Reply:
You are right. But we can't set the release to null as it means that the array is invalid.
I have to create another release callback which don't delete the buffers